research-software-directory/RSD-as-a-service
# Replace unimported with knip

# Replace unimported with knip

Closes #1273

Changes proposed in this pull request:
*  Replace "retired" [unimported](https://github.com/smeijer/unimported) with [knip](https://knip.dev/).

How to test:
* This is specific frontend "feature" to check for unimported files, unused types, unused dependencies and some other "quality" checks.  It is not integrated in the CI/CD, rather it is executed manually from time to time. **To test you will need to have Node and NPM installed on your machine.**
* Navigate to frontend folder and run `npm install` to install all dependencies. You might also remove old node_modules folder using `rm -rf node_modules` **before running** `npm install`
* `npm run knip:files` to check for unimported/unused files
* `npm run knip:types` to check for unused exported type definitions
* Both checks should produce 0 in this PR because I removed unused files and types.
* `npm test` to run unit tests locally. All test should pass.
* `npm run build` to build frontend. The build should be successful.
* `make start` to build app and create test data. The app should work without problems.

PR Checklist:
